FindmallocMC.cmake

```diff
@@ -97,44 +102,47 @@ if(mallocMC_ROOT_DIR)
     # mallocMC variables ########################################################
     #
     set(mallocMC_VERSION "${mallocMC_VERSION_MAJOR}.${mallocMC_VERSION_MINOR}.${mallocMC_VERSION_PATCH}")
-    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include)
+    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include CACHE PATH "mallocMC include pathes")
 
     # check additional components ###############################################
     #
     foreach(COMPONENT ${mallocMC_FIND_COMPONENTS})
-        set(mallocMC_${COMPONENT}_FOUND TRUE)
+        set(mallocMC_${COMPONENT}_FOUND TRUE CACHE BOOL
+            "The mallocMC component ${COMPONENT} exists in the found installation.")
 
         if(${COMPONENT} STREQUAL "halloc")
 
             # halloc linked library #################################################
             #
             list(APPEND mallocMC_REQUIRED_VARS_LIST mallocMC_LIBRARIES)
-            find_library(${COMPONENT}_LIBRARY
+            find_library(mallocMC_${COMPONENT}_LIBRARY
                 NAMES libhalloc.a
                 PATHS "${mallocMC_ROOT_DIR}/../halloc/" ENV HALLOC_ROOT
                 PATH_SUFFIXES "lib" "bin"
-                )
-            if(${COMPONENT}_LIBRARY)
-                list(APPEND mallocMC_LIBRARIES ${${COMPONENT}_LIBRARY})
-            else(${COMPONENT}_LIBRARY)
+                DOC "Libraries for the mallocMC component ${COMPONENT}."
+            )
+            if(mallocMC_${COMPONENT}_LIBRARY)
+                list(APPEND mallocMC_LIBRARIES ${mallocMC_${COMPONENT}_LIBRARY})
```

Comment:
It seems that `list(APPEND)` can not directly create cached variables.
We need to use `set(mallocMC_LIBRARIES ${mallocMC_LIBRARIES} ${mallocMC_${COMPONENT}_LIBRARY} CACHE PATH "Libraries for the mallocMC compontent ${COMPONENT}")

Otherwise, the mallocMC_LIBRARIES variable is not visible within ccmake